### PR TITLE
mgr/cephadm/schedule: fix message

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by tchaikov** [comment]
cc @huww98

**Comment by huww98** [comment]
Following up https://github.com/ceph/ceph/commit/1bf09d143990c5199f71a28c88d6e591b6fbd934#r50566843 .

Still not making sense to me. If some host does not belong to mon public_network, then why will this happen when upgrading, since I don't change any IP address? And why it is resolved by itself?

**Comment by liewegas** [comment]
> Following up [1bf09d1#r50566843](https://github.com/ceph/ceph/commit/1bf09d143990c5199f71a28c88d6e591b6fbd934#r50566843) .
> 
> Still not making sense to me. If some host does not belong to mon public_network, then why will this happen when upgrading, since I don't change any IP address? And why it is resolved by itself?

I'm a bit confused by that as well.  My best guess is that you'll see the same message again if you restart the mgr daemon or do 'ceph orch apply mon 5' (or whatever the current placement is)

File: src/pybind/mgr/cephadm/schedule.py [hunk 1]
```diff
@@ -304,7 +304,7 @@ def get_candidates(self) -> List[DaemonPlacement]:
                     ls.append(h)
                 else:
                     logger.info(
-                        f"Filtered out host {h.hostname}: could not verify host allowed virtual ips")
+                        f"Filtered out host {h.hostname}: host does not belong to mon public_network")
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/schedule.py:307]: we're trying to invent a message here, but only `self.filter_new_host` can provide a proper message. can we move this this log message to the implementation of self.filter_new_host ?

**Comment by huww98** [comment]
> I'm a bit confused by that as well. My best guess is that you'll see the same message again if you restart the mgr daemon or do 'ceph orch apply mon 5' (or whatever the current placement is)

No, the log looks like this:
```
cephadm 2021-05-10T14:06:21.298582+0000 mgr.gpu024.bapbcz (mgr.6832120) 96 : cephadm [INF] Schedule restart daemon mgr.gpu024.bapbcz
cluster 2021-05-10T14:06:33.512186+0000 mon.gpu023 (mon.0) 27177 : cluster [INF] Active manager daemon gpu024.bapbcz restarted
cluster 2021-05-10T14:06:33.514667+0000 mon.gpu023 (mon.0) 27178 : cluster [INF] Activating manager daemon gpu024.bapbcz
cluster 2021-05-10T14:06:33.627520+0000 mon.gpu023 (mon.0) 27215 : cluster [INF] Manager daemon gpu024.bapbcz is now available
cephadm 2021-05-10T14:06:44.939369+0000 mgr.gpu024.bapbcz (mgr.6832210) 8 : cephadm [INF] Saving service mon spec with placement cpu0;gpu023;gpu024;gpu006;gpu013
```

The logs when upgrading looks like:
```
cephadm 2021-05-08T17:03:59.678460+0000 mgr.gpu014.kwbqcf (mgr.6664231) 490838 : cephadm [INF] Upgrade: Need to upgrade myself (mgr.gpu014.kwbqcf)
cephadm 2021-05-08T17:03:59.684683+0000 mgr.gpu014.kwbqcf (mgr.6664231) 490839 : cephadm [INF] Failing over to other MGR
cluster 2021-05-08T17:04:04.752923+0000 mon.cpu0 (mon.0) 192255 : cluster [INF] Active manager daemon gpu024.bapbcz restarted
cluster 2021-05-08T17:04:04.761191+0000 mon.cpu0 (mon.0) 192256 : cluster [INF] Activating manager daemon gpu024.bapbcz
cluster 2021-05-08T17:04:05.508183+0000 mon.cpu0 (mon.0) 192259 : cluster [INF] Manager daemon gpu024.bapbcz is now available
cephadm 2021-05-08T17:04:12.750241+0000 mgr.gpu024.bapbcz (mgr.6727372) 4 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:12.750354+0000 mgr.gpu024.bapbcz (mgr.6727372) 5 : cephadm [INF] Filtered out host gpu023: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:12.750443+0000 mgr.gpu024.bapbcz (mgr.6727372) 6 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:12.750525+0000 mgr.gpu024.bapbcz (mgr.6727372) 7 : cephadm [INF] Filtered out host gpu006: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:12.750603+0000 mgr.gpu024.bapbcz (mgr.6727372) 8 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:12.762441+0000 mgr.gpu024.bapbcz (mgr.6727372) 9 : cephadm [INF] Upgrade: Need to upgrade myself (mgr.gpu024.bapbcz)
cephadm 2021-05-08T17:04:13.272071+0000 mgr.gpu024.bapbcz (mgr.6727372) 10 : cephadm [INF] Upgrade: Pulling cr.scut-smil.cn/infra/ceph:v16.2.3 on gpu014
cephadm 2021-05-08T17:04:36.641321+0000 mgr.gpu024.bapbcz (mgr.6727372) 23 : cephadm [INF] Upgrade: Updating mgr.gpu014.kwbqcf
cephadm 2021-05-08T17:04:36.681668+0000 mgr.gpu024.bapbcz (mgr.6727372) 24 : cephadm [INF] Deploying daemon mgr.gpu014.kwbqcf on gpu014
cluster 2021-05-08T17:04:58.494557+0000 mon.gpu023 (mon.1) 24900 : cluster [INF] mon.gpu023 calling monitor election
cluster 2021-05-08T17:04:58.510456+0000 mon.gpu013 (mon.0) 31261 : cluster [INF] mon.gpu013 calling monitor election
cluster 2021-05-08T17:04:58.511531+0000 mon.gpu006 (mon.3) 147 : cluster [INF] mon.gpu006 calling monitor election
cluster 2021-05-08T17:05:03.533817+0000 mon.gpu013 (mon.0) 31262 : cluster [INF] mon.gpu013 is new leader, mons gpu013,gpu023,gpu006 in quorum (ranks 0,1,3)
cephadm 2021-05-08T17:04:58.431465+0000 mgr.gpu024.bapbcz (mgr.6727372) 36 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:58.431548+0000 mgr.gpu024.bapbcz (mgr.6727372) 37 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:58.431613+0000 mgr.gpu024.bapbcz (mgr.6727372) 38 : cephadm [INF] Filtered out host gpu006: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:58.431671+0000 mgr.gpu024.bapbcz (mgr.6727372) 39 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:04:58.441150+0000 mgr.gpu024.bapbcz (mgr.6727372) 40 : cephadm [INF] Safe to remove mon.cpu0: new quorum should be ['gpu013', 'gpu023', 'gpu024', 'gpu006'] (from ['gpu013', 'gpu023', 'gpu024', 'gpu006'])
cephadm 2021-05-08T17:04:58.441290+0000 mgr.gpu024.bapbcz (mgr.6727372) 41 : cephadm [INF] Removing monitor cpu0 from monmap...
cephadm 2021-05-08T17:04:58.446958+0000 mgr.gpu024.bapbcz (mgr.6727372) 42 : cephadm [INF] Removing daemon mon.cpu0 from cpu0
cluster 2021-05-08T17:05:00.468861+0000 mon.gpu024 (mon.2) 36 : cluster [INF] mon.gpu024 calling monitor election
cephadm 2021-05-08T17:05:01.883890+0000 mgr.gpu024.bapbcz (mgr.6727372) 47 : cephadm [INF] Reconfiguring crash.cpu0 (monmap changed)...
cluster 2021-05-08T17:05:05.493338+0000 mon.gpu013 (mon.0) 31373 : cluster [INF] mon.gpu013 calling monitor election
cluster 2021-05-08T17:05:05.497049+0000 mon.gpu023 (mon.1) 24901 : cluster [INF] mon.gpu023 calling monitor election
cluster 2021-05-08T17:05:05.518236+0000 mon.gpu013 (mon.0) 31374 : cluster [INF] mon.gpu013 is new leader, mons gpu013,gpu023,gpu024,gpu006 in quorum (ranks 0,1,2,3)
cluster 2021-05-08T17:05:05.526561+0000 mon.gpu013 (mon.0) 31379 : cluster [INF] Health check cleared: MON_DOWN (was: 1/4 mons down, quorum gpu013,gpu023,gpu006)
cephadm 2021-05-08T17:05:05.643759+0000 mgr.gpu024.bapbcz (mgr.6727372) 50 : cephadm [INF] Reconfiguring daemon crash.cpu0 on cpu0
...
cephadm 2021-05-08T17:07:50.304523+0000 mgr.gpu014.kwbqcf (mgr.6750496) 177 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:07:50.304893+0000 mgr.gpu014.kwbqcf (mgr.6750496) 178 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:07:50.304981+0000 mgr.gpu014.kwbqcf (mgr.6750496) 179 : cephadm [INF] Filtered out host gpu006: could not verify host allowed virtual ips
cephadm 2021-05-08T17:07:50.305078+0000 mgr.gpu014.kwbqcf (mgr.6750496) 180 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:07:50.320808+0000 mgr.gpu014.kwbqcf (mgr.6750496) 181 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:08:05.337629+0000 mgr.gpu014.kwbqcf (mgr.6750496) 189 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:08:20.352143+0000 mgr.gpu014.kwbqcf (mgr.6750496) 198 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:08:35.369326+0000 mgr.gpu014.kwbqcf (mgr.6750496) 206 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:08:51.415507+0000 mgr.gpu014.kwbqcf (mgr.6750496) 215 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:08:51.415706+0000 mgr.gpu014.kwbqcf (mgr.6750496) 216 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:08:51.415872+0000 mgr.gpu014.kwbqcf (mgr.6750496) 217 : cephadm [INF] Filtered out host gpu006: could not verify host allowed virtual ips
cephadm 2021-05-08T17:08:51.415991+0000 mgr.gpu014.kwbqcf (mgr.6750496) 218 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:08:51.434103+0000 mgr.gpu014.kwbqcf (mgr.6750496) 219 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:09:06.449302+0000 mgr.gpu014.kwbqcf (mgr.6750496) 228 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:09:21.466434+0000 mgr.gpu014.kwbqcf (mgr.6750496) 236 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
cephadm 2021-05-08T17:09:36.476534+0000 mgr.gpu014.kwbqcf (mgr.6750496) 245 : cephadm [INF] Upgrade: It is NOT safe to stop mon.gpu023 at this time: not enough monitors would be available (gpu024) after stopping mons [gpu023]
...(repeat many times)
cephadm 2021-05-08T17:18:16.678844+0000 mgr.gpu014.kwbqcf (mgr.6750496) 571 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:16.678988+0000 mgr.gpu014.kwbqcf (mgr.6750496) 572 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:16.679118+0000 mgr.gpu014.kwbqcf (mgr.6750496) 573 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:16.685029+0000 mgr.gpu014.kwbqcf (mgr.6750496) 574 : cephadm [INF] Deploying daemon mon.gpu006 on gpu006
cluster 2021-05-08T17:18:23.970362+0000 mon.gpu024 (mon.1) 188 : cluster [INF] mon.gpu024 calling monitor election
cluster 2021-05-08T17:18:23.984279+0000 mon.gpu023 (mon.0) 26751 : cluster [INF] mon.gpu023 calling monitor election
cluster 2021-05-08T17:18:29.005961+0000 mon.gpu023 (mon.0) 26761 : cluster [INF] mon.gpu023 is new leader, mons gpu023,gpu024 in quorum (ranks 0,1)
cephadm 2021-05-08T17:18:29.228513+0000 mgr.gpu014.kwbqcf (mgr.6750496) 581 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:29.234000+0000 mgr.gpu014.kwbqcf (mgr.6750496) 582 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:29.235023+0000 mgr.gpu014.kwbqcf (mgr.6750496) 583 : cephadm [INF] Filtered out host gpu013: could not verify host allowed virtual ips
cephadm 2021-05-08T17:18:29.257156+0000 mgr.gpu014.kwbqcf (mgr.6750496) 584 : cephadm [INF] Upgrade: It appears safe to stop mon.gpu006
cluster 2021-05-08T17:18:25.964655+0000 mon.gpu006 (mon.2) 1 : cluster [INF] mon.gpu006 calling monitor election
cluster 2021-05-08T17:18:30.988246+0000 mon.gpu006 (mon.2) 2 : cluster [INF] mon.gpu006 calling monitor election
cluster 2021-05-08T17:18:30.989345+0000 mon.gpu024 (mon.1) 189 : cluster [INF] mon.gpu024 calling monitor election
cluster 2021-05-08T17:18:31.002518+0000 mon.gpu023 (mon.0) 26883 : cluster [INF] mon.gpu023 calling monitor election
cluster 2021-05-08T17:18:31.030779+0000 mon.gpu023 (mon.0) 26886 : cluster [INF] mon.gpu023 is new leader, mons gpu023,gpu024,gpu006 in quorum (ranks 0,1,2)
cluster 2021-05-08T17:18:31.039878+0000 mon.gpu023 (mon.0) 26891 : cluster [INF] Health check cleared: MON_DOWN (was: 1/3 mons down, quorum gpu023,gpu024)
cephadm 2021-05-08T17:18:59.654067+0000 mgr.gpu014.kwbqcf (mgr.6750496) 602 : cephadm [INF] Upgrade: Updating mon.gpu006
cephadm 2021-05-08T17:18:59.670067+0000 mgr.gpu014.kwbqcf (mgr.6750496) 603 : cephadm [INF] Deploying daemon mon.gpu006 on gpu006
cluster 2021-05-08T17:19:04.118763+0000 mon.gpu006 (mon.2) 1 : cluster [INF] mon.gpu006 calling monitor election
cluster 2021-05-08T17:19:04.121041+0000 mon.gpu024 (mon.1) 190 : cluster [INF] mon.gpu024 calling monitor election
cluster 2021-05-08T17:19:04.128956+0000 mon.gpu023 (mon.0) 27065 : cluster [INF] mon.gpu023 calling monitor election
cluster 2021-05-08T17:19:04.149828+0000 mon.gpu023 (mon.0) 27066 : cluster [INF] mon.gpu023 is new leader, mons gpu023,gpu024,gpu006 in quorum (ranks 0,1,2)
cephadm 2021-05-08T17:19:10.996242+0000 mgr.gpu014.kwbqcf (mgr.6750496) 609 : cephadm [INF] Filtered out host cpu0: could not verify host allowed virtual ips
cephadm 2021-05-08T17:19:10.996423+0000 mgr.gpu014.kwbqcf (mgr.6750496) 610 : cephadm [INF] Filtered out host gpu024: could not verify host allowed virtual ips
cephadm 2021-05-08T17:19:11.001299+0000 mgr.gpu014.kwbqcf (mgr.6750496) 611 : cephadm [INF] Deploying daemon mon.gpu013 on gpu013
...
```
This message first appears when a new mgr is activated. Then This checking is preventing new mon from being deployed, and hindering the upgrade process of mons. After about 10 minutes, at 17:18:16, host `gpu006` is no longer being filtered out, and finally mon can be deployed on it.

I guess when a new manager starts, it does not have enough information. So it just prevents deploying for safe. But what blocks it for 10 minutes?

**Comment by liewegas** [comment]
Oh! I know what the problem is.  1897d1cd15af385bd888da0a9ee944cd3a68af07 changed the way we store the per-host network interface/network info.  On upgrade, cephadm thinks there are no networks on each host until the device refresh happens.

**Review by sebastian-philipp — APPROVED** [review]

**Comment by tchaikov** [comment]
- https://pulpito.ceph.com/kchai-2021-05-12_04:48:43-rados-wip-kefu-testing2-2021-05-12-1103-distro-basic-smithi/
- https://pulpito.ceph.com/kchai-2021-05-13_03:34:40-rados-wip-kefu-testing-2021-05-12-2313-distro-basic-smithi/

**Pull request merged**